Ember Octane refactor

RKennedy9064 opened the pull request:
@bart-degreed I had some time so I made an initial refactor for Ember Octane. I've updated almost everything except `todo-item.js` because it looks like it isn't compatible with Ember Octane, so I left it with the old model definition. It might not cover everything, but I figure it's a decent start. Let me know what you think.

RKennedy9064 committed:
Ember Octane refactor

bart-degreed commented on TodoListClient/app/routes/s/todo-items/add.js:
```diff
@@ -1,19 +1,10 @@
-import DataRoute from 'ember-data-route';
+import { inject as service } from '@ember/service';
+
+export default class IndexRoute extends Route {
```
Should this class be named AddRoute instead?

RKennedy9064 replied:
Good catch, I copy pasted the index route and forgot to rename it.

bart-degreed reviewed (changes requested):
Thanks a lot for your help, this is awesome! I've tested locally and everything seems to work without problems.

I do have a few questions, though.
- When I run "yarn install", it wildly changes yarn.lock and it changes the top-level comment back to:
    ```
    # THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
    ```
    so should I run something else to get the same file from your PR?
- running "yarn lint:fix" still produces a few errors and it changes whitespace in various files. Can you take a look at that? I don't have a preference for a coding style in these files, but it would be nice if it matches the configured style settings, so feel free to update as you feel is most appropriate.
- Testing in Chrome still produces several deprecation warnings on the console. Is that something you can fix?
- Same for various deprecation warnings that are shown when running "ember s".

RKennedy9064 commented:
@bart-degreed I should have some time this weekend to finish those clean ups. I think the yarn error was because I was playing around with Yarn 2, which doesn't work with ember. When I rebuilt with Yarn 1 I got the same auto generated file line you did.

Also one of the lints has to do with declaring the todo item class. I don't think ember-cp-validations is compatible with the new model syntax at this time. Is using ember-cp-validations set in stone, or are you open to switching to a validation library that supports the new syntax?

bart-degreed commented:
> Is using ember-cp-validations set in stone, or are you open to switching to a validation library that supports the new syntax?

Please use what you believe makes most sense today in the Ember ecosystem. This applies to other dependencies too. I tried to update simple-auth to latest syntax as described on its GitHub readme, but got stuck.

RKennedy9064 committed:
yarn lint:fix

RKennedy9064 committed:
Added ember-changeset-validations

RKennedy9064 committed:
Super basic UI

RKennedy9064 committed:
Cleanup deprecation

RKennedy9064 committed:
yarn lint:fix

RKennedy9064 committed:
Add autocomplete suggestions

RKennedy9064 commented:
@bart-degreed Okay I just posted my newest changes. I added some basic stylings, if you don't like them I can revert that commit. The bootstrap dependency I added has 1 deprecation warning, but I created a PR for fixing that [here](https://github.com/kybishop/ember-popper/pull/140) that should hopefully fix that soon.

I still see a few deprecation messages about injecting session and store, but I think those might be from other addons? Everywhere in the app where I use session and store I explicitly inject it and ember inspector doesn't show any deprecations, so I'm not really sure what to think of those last 2 deprecations in the Chrome console.

RKennedy9064 committed:
Rename route

RKennedy9064 commented:
Also did some digging on the ember discord and I think the remaining deprecations are from ember-simple-auth. There's a few issues where they're using deprecated code, so that should hopefully resolve itself once they fix it and update. https://github.com/simplabs/ember-simple-auth/issues/2309

bart-degreed commented on TodoListClient/app/components/forms/todo-item-form/component.js:
```diff
@@ -1,9 +1,44 @@
+  constructor() {
+    super(...arguments);
+
+    // Was unable to use the changeset helpr, so I'm constructing the changeset manually.
+    // Most likely related to this issue, but the fix didn't work.
+    // https://github.com/poteto/ember-changeset-validations/issues/214
```
I appreciate how you've gone to great lengths to get this working. The thing is that TodoListClient is intended to showcase how basic Ember features can be used. Therefore I'm not a fan of putting in this workaround unless it is temporary. Can you open a new issue at ember-changeset-validations, to get this sorted out? I've subscribed to the other linked issues from your PR.

RKennedy9064 replied:
I don't really like constructing the variable manually either, so I'll look into this/open an issue. I'm hoping I can play around with it and get it to work as expected.

I also removed Chrome. I downloaded since some unit tests in another module I made a PR for needed it and I think my browser just remembered my last download path where I put the logo.

bart-degreed commented on TodoListClient/app/models/todo-item.js:
```diff
@@ -1,18 +1,6 @@
-  owner: belongsTo('person')
-
-});
+export default class TodoItem extends Model {
```
Should this be named TodoItemModel, similar to PersonModel?

bart-degreed commented on TodoListClient/app/routes/application.js:
```diff
@@ -1,9 +1,11 @@
+  @service session;
+  @service router;
+
+  beforeModel(/* transition */) {
```
why this comment?

RKennedy9064 replied:
It was mainly just to show that the transition is an available variable without the linter complaining that it wasn't used, but I can remove it.

bart-degreed reviewed (changes requested):
I love what you've done to the project! It looks great now, still works correctly, and fixes most warnings!

Few remarks, see below.
Something else: why was ChromeSetup.exe added to git?

RKennedy9064 committed:
Remove Chrome

RKennedy9064 committed:
Remove comment

RKennedy9064 committed:
Rename TodoItem

RKennedy9064 committed:
Remove warning about missing serializer file

bart-degreed reviewed (approved):
Thanks for processing feedback. I think this can be merged now, but I prefer to wait a bit and see if you're able to solve the validation problem. Or at least link to an open issue from there, so we can remove the workaround in vNext.

bart-degreed commented:
@RKennedy9064 are you still working on this?

RKennedy9064 commented:
@bart-degreed I got caught up with a few work and real life related things, so I haven't had time to finish this. I'm hoping to have some free time this weekend to finish up the remaining tasks.

bart-degreed commented:
Great, thanks.

RKennedy9064 commented:
@bart-degreed Quick update on this. I've been super busy with work, but on the plus side I'm currently upgrading a project that uses ember-changeset-validations, so I'm working on the same issue as this PR. Hopefully once I get it solved I can just apply what I used for the project I'm currently working on and update this PR.

bart-degreed commented:
What a nice coincidence. Thanks for the update.
